recipes/boost/all/conanfile.py

```diff
@@ -588,6 +588,10 @@ def disable_cobalt():
                 elif not self._has_cppstd_14_supported:
                     disable_graph()
 
+            if self.settings.os == "iOS":
```

Comment:
That's true - however, when trying it out, i ran into some further issues with watchos and tvos:

- libbacktrace doesn't build for those platforms, so i needed `-oboost:with_stacktrace_backtrace=False`
- libiconv doesn't build for those platforms, so i needed `-oboost:without_locale=True`
- Boost.Test doesn't build (it's trying to use `sigaltstack`, which isn't available on them), so i needed `-oboost:without_test=True`

libbacktrace and libiconv both had the same issue: autoconf didn't know about watchos or tvos.

It's not clear to me which of these options should be encoded into the conanfile as defaults - eg: if libiconv or libbacktrace gets fixed, it will (in theory) suddenly be possible to build Boost without disabling those options.

Reply:
Please, let's keep focused on iOS only. The Boost recipe should be surgical changed and well tested for any change.

Reply:
Thanks for taking the time to test this on other platforms!

Interesting observation about backtrace and iconv - I'd had expected roughly that most things that build for iOS also build for watchOS and tvOS.
Given that there's further testing pending on those - perhaps we just for users targeting those platforms to report back.